test/conftest.py

```diff
@@ -20,3 +23,33 @@ def test_dir(tmp_path_factory):
 def play_scraper(icon_dir, test_dir):
     base_url = pytest.FILE + str(test_dir) + "/"
     return PlayAppPageScraper(base_url, icon_dir)
+
+
+@pytest.fixture
+def input_dir(tmp_path_factory):
+    return tmp_path_factory.mktemp("input_dir")
+
+
+@pytest.fixture
+def icons(input_dir):
```

Comment:
In general it's good practice to have factories to build your test data inside of each test case instead of hardcoding a specific set of data. In your case the factory could have a constructor like this:

```python
class IconBuilder:
    def __init__(self, num_icons: int, num_categories:Optional[int]=None, image_size:(int, int)):
        pass
```